src/features/dashboard/components/Dialogs/DeleteAppDialog/index.tsx

```diff
@@ -10,52 +11,47 @@ type TDeleteAppDialogProps = {
 
 const DeleteAppDialog = ({ appId, onClose }: TDeleteAppDialogProps) => {
   const { deleteApp } = useDeleteApp();
+  const { deviceType } = useDeviceType();
+  const [isMobile, setIsMobile] = useState(deviceType === 'mobile');
 
   const onOpenChange = useCallback(
-    (open: boolean) => {
+    (open) => {
       if (!open) {
         onClose();
       }
     },
     [onClose],
   );
 
-  const actionButtons: TModalActionButton[] = useMemo(
-    () => [
-      {
-        id: 0,
-        text: 'Yes, delete',
-        color: 'primary',
-        onClick: () => {
-          deleteApp(appId);
-          onClose();
-        },
-      },
-      {
-        id: 1,
-        text: 'Cancel',
-        color: 'secondary',
-        onClick: () => {
-          onClose();
-        },
-      },
-    ],
-    [appId, deleteApp, onClose],
-  );
+  useEffect(() => {
+    setIsMobile(deviceType === 'mobile');
+  }, [deviceType]);
 
   return (
-    <Modal defaultOpen onOpenChange={onOpenChange}>
-      <Modal.Portal>
-        <div className='modal-overlay'>
-          <Modal.Overlay />
-          <Modal.DialogContent
-            title='Delete app'
-            content='Are you sure you want to delete this app?'
-            action_buttons={actionButtons}
-            has_close_button
-          />
+    <Modal
+      isOpened={true}
+      toggleModal={onOpenChange}
+      primaryButtonLabel='Yes, delete'
+      secondaryButtonLabel='Cancel'
+      disableCloseOnOverlay
+      isMobile={isMobile}
```

Comment:
@agrim-deriv no need to add useEffect for setting the `setIsMobile`. We can directly pass this condition here `isMobile={deviceType !== 'desktop')`

Reply:
This issue has been fixed @shafin-deriv